src/adapters/refract/transformSections.coffee

```diff
@@ -2,37 +2,65 @@ _ = require('./helper')
 
 blueprintApi = require('../../blueprint-api')
 getDescription = require('./getDescription')
+
 transformResources = require('./transformResources')
+transformResource = require('./transformResource')
+
 
-module.exports = (element) ->
+module.exports = (parentElement) ->
   resourceGroups = []
-  resourceGroupsElements = _.chain(element)
-                    .content()
-                    .filter({element: 'category', meta: {classes: ['resourceGroup']}})
-                    .value()
-
-  # If we can't find any resource group we can assume that there are Resources directly
-  if resourceGroupsElements.length is 0
-    resourceGroups.push(
-      new blueprintApi.Section({
-        name: ''
+
+  # List of Application AST Resources (= already transformed
+  # Refract Resources into the Application AST Resource).
+  applicationAstResources = []
+
+  _.forEach(parentElement.content, (element, index) ->
+    # If the element is a resource, transform it
+    # into the Application AST Resource and push it
+    # to a temporary array, we'll create an artificial
+    # resource group (section) for these resources
+    # later in the process.
+    if element.element is 'resource'
+      applicationAstResources = applicationAstResources.concat(
+        transformResource(element)
+      )
+
+    # Element is category (resource group most probably).
+    if element.element is 'category'
+      # First let's create an artificial resource group (section)
+      # for previous resources.
+      if applicationAstResources.length
+        resourceGroups.push(new blueprintApi.Section({
+          name: ''
+          resources: applicationAstResources
+        }))
+
+        applicationAstResources = []
```

Comment:
So I would recommend a couple of things. Up to you if you would like to do :)

1. Rename `applicationAstResources` to `resourcesWithoutGroup`
1. Add a comment when you "slurp" up these resources and reset to empty, because you're putting them into a group

This will then give a little more information about the logic and why it's going on. Just some thoughts :)